sormas-ui/src/main/java/de/symeda/sormas/ui/contact/MergeContactsGrid.java

```diff
@@ -0,0 +1,292 @@
+package de.symeda.sormas.ui.contact;
+
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Stream;
+
+import com.vaadin.data.TreeData;
+import com.vaadin.data.provider.TreeDataProvider;
+import com.vaadin.icons.VaadinIcons;
+import com.vaadin.server.ExternalResource;
+import com.vaadin.server.Page;
+import com.vaadin.ui.Button;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.Label;
+import com.vaadin.ui.Link;
+import com.vaadin.ui.Notification;
+import com.vaadin.ui.Notification.Type;
+import com.vaadin.ui.StyleGenerator;
+import com.vaadin.ui.TreeGrid;
+import com.vaadin.ui.renderers.DateRenderer;
+import com.vaadin.ui.renderers.TextRenderer;
+import com.vaadin.ui.themes.ValoTheme;
+
+import de.symeda.sormas.api.DiseaseHelper;
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.Language;
+import de.symeda.sormas.api.caze.AgeAndBirthDateDto;
+import de.symeda.sormas.api.caze.CaseIndexDto;
+import de.symeda.sormas.api.contact.ContactCriteria;
+import de.symeda.sormas.api.contact.ContactIndexDto;
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.person.PersonHelper;
+import de.symeda.sormas.api.utils.DataHelper;
+import de.symeda.sormas.api.utils.DateHelper;
+import de.symeda.sormas.ui.SormasUI;
+import de.symeda.sormas.ui.utils.ButtonHelper;
+import de.symeda.sormas.ui.utils.CssStyles;
+import de.symeda.sormas.ui.utils.VaadinUiUtil;
+
+@SuppressWarnings("serial")
+public class MergeContactsGrid extends TreeGrid<ContactIndexDto> {
+
+	public static final String COLUMN_DISEASE = Captions.columnDiseaseShort;
+	public static final String COLUMN_ACTIONS = "actions";
+	public static final String COLUMN_COMPLETENESS = "completenessValue";
+	public static final String COLUMN_UUID = "uuidLink";
+
+	private ContactCriteria criteria;
+	private boolean ignoreRegion;
+
+	private List<String[]> hiddenUuidPairs;
+
+	@SuppressWarnings("unchecked")
+	public MergeContactsGrid() {
+
+		super(ContactIndexDto.class);
+		setSizeFull();
+		setSelectionMode(SelectionMode.NONE);
+
+		Column<ContactIndexDto, String> diseaseColumn = addColumn(contact -> DiseaseHelper.toString(contact.getDisease(), contact.getDiseaseDetails()));
+		diseaseColumn.setId(COLUMN_DISEASE);
+
+		addComponentColumn(indexDto -> {
+			return buildButtonLayout(indexDto);
+		}).setId(COLUMN_ACTIONS);
+
+		addComponentColumn(indexDto -> {
```

Comment:
it would be nice to add an abstraction between this class and the TreeGrid where the common parts of the merge grid classes can go